web/libs/ui/src/lib/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable-next-line */
+import React, { type InputHTMLAttributes } from "react";
+import styles from "./checkbox.module.scss";
+import { nanoid } from "nanoid";
+
+export interface CheckboxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "type"> {
+  indeterminate?: boolean;
+}
+const classNameMap = { ...styles };
+
+export const Checkbox = ({ checked, indeterminate, style, onChange, children, ...props }: CheckboxProps) => {
+  const checkboxRef = React.createRef();
+  const withLabel = !!children;
+  const [id] = React.useState(nanoid);
+
+  React.useEffect(() => {
+    if (checkboxRef.current) {
+      checkboxRef.current.indeterminate = indeterminate;
+    }
+  }, [checkboxRef, indeterminate]);
+
+  const checkboxContent = (
+    <span className={styles.checkbox__box}>
+      <input
+        {...props}
+        id={id}
+        ref={checkboxRef as React.Ref<HTMLInputElement>}
+        checked={!!checked}
+        className={styles.checkbox__input}
+        type="checkbox"
+        onChange={(e) => {
+          onChange?.(e);
+        }}
+      />
+      <span
+        className={`${styles.checkbox__check} ${checked ? styles.checkbox_checked : ""} ${
+          indeterminate ? styles.checkbox_indeterminate : ""
+        }`}
```

Comment:
we definitely need to use some simple library to work with classnames. even without it we can make it more readable:
```suggestion
        className={[
          styles.checkbox__check,
          checked && styles.checkbox_checked,
          indeterminate && styles.checkbox_indeterminate,
        ].filter(Boolean).join(" ")}
```

Reply:
Yes definitely going to happen, something along the lines of:

https://github.com/lukeed/clsx

or

https://cva.style/docs

Reply:
For now, what you suggested is a great lightweight alternative, especially as this component does not have complex needs.

Reply:
yep, `clsx` looks like it!